tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeKLibraryImpl.kt

```diff
@@ -0,0 +1,65 @@
+package org.jetbrains.kotlin.gradle.plugin.experimental.internal
+
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.ConfigurationContainer
+import org.gradle.api.attributes.AttributeContainer
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.internal.component.SoftwareComponentInternal
+import org.gradle.api.internal.component.UsageContext
+import org.gradle.api.internal.file.FileOperations
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.language.cpp.internal.DefaultUsageContext
+import org.gradle.nativeplatform.Linkage
+import org.jetbrains.kotlin.gradle.plugin.experimental.KotlinNativeKLibrary
+import org.jetbrains.kotlin.gradle.plugin.experimental.sourcesets.KotlinNativeSourceSet
+import org.jetbrains.kotlin.konan.target.CompilerOutputKind
+import javax.inject.Inject
+
+open class KotlinNativeKLibraryImpl @Inject constructor(
+        name: String,
+        baseName: Provider<String>,
+        componentImplementation: Configuration,
+        sources: KotlinNativeSourceSet,
+        identity: KotlinNativeVariantIdentity,
+        objects: ObjectFactory,
+        projectLayout: ProjectLayout,
+        configurations: ConfigurationContainer,
+        fileOperations: FileOperations
+) : KotlinNativeBinaryImpl(name,
+        baseName,
+        sources,
+        identity,
+        projectLayout,
+        CompilerOutputKind.LIBRARY,
+        objects,
+        componentImplementation,
+        configurations,
+        fileOperations),
+        KotlinNativeKLibrary,
+    SoftwareComponentInternal
```

Comment:
Same header issue like above

Reply:
Consider to add the TODO here as well..
Otherwise it may be unseen if someone refactore the other class...

Reply:
Fixed.